src/nvim/fileio.c

```diff
@@ -2781,20 +2781,12 @@ buf_write (
     else
       backup_ext = p_bex;
 
-    if (backup_copy && (fd = os_open((char *)fname, O_RDONLY, 0)) >= 0) {
-      int bfd;
-      char_u      *copybuf, *wp;
-      int some_error = FALSE;
+    if (backup_copy) {
```

Comment:
@Said6289 I just noticed: the old code checked `os_open` result, and falls back to the `else` branch if it fails. I guess that might be worth keeping, although we have no use for the file-descriptor.

Reply:
I guess we can use a stat for that, so we don't have an open fd lying around.